CMakeLists.txt

```diff
@@ -0,0 +1,423 @@
+rapids_cpm_init()
+
+if(Legion_USE_CUDA)
+  include(cmake/thirdparty/get_nccl.cmake)
+  include(cmake/thirdparty/get_thrust.cmake)
+endif()
+
+if(Legion_USE_Python)
+  find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
+  message(STATUS "legate.core: Has Python3: ${Python3_FOUND}")
+  message(STATUS "legate.core: Has Python 3 interpreter: ${Python3_Interpreter_FOUND}")
+  message(STATUS "legate.core: Python 3 include directories: ${Python3_INCLUDE_DIRS}")
+  message(STATUS "legate.core: Python 3 libraries: ${Python3_LIBRARIES}")
+  message(STATUS "legate.core: Python 3 library directories: ${Python3_LIBRARY_DIRS}")
+  message(STATUS "legate.core: Python 3 version: ${Python3_VERSION}")
+  set(Legion_Python_Version ${Python3_VERSION})
+endif()
+
+include(cmake/thirdparty/get_legion.cmake)
+
+##############################################################################
+# - legate.core --------------------------------------------------------------
+
+set(LEGATE_CORE_SOURCES "")
+set(LEGATE_CORE_CXX_DEFS "")
+set(LEGATE_CORE_CUDA_DEFS "")
+set(LEGATE_CORE_CXX_OPTIONS "")
+set(LEGATE_CORE_CUDA_OPTIONS "")
+
+if(NOT CMAKE_BUILD_TYPE STREQUAL "Release")
+  list(APPEND LEGATE_CORE_CXX_DEFS DEBUG_LEGATE)
+  list(APPEND LEGATE_CORE_CUDA_DEFS DEBUG_LEGATE)
+endif()
+
+if(Legion_USE_CUDA)
+  list(APPEND LEGATE_CORE_CXX_DEFS LEGATE_USE_CUDA)
+  list(APPEND LEGATE_CORE_CUDA_DEFS LEGATE_USE_CUDA)
+endif()
+
+if(Legion_USE_OpenMP)
+  list(APPEND LEGATE_CORE_CXX_DEFS LEGATE_USE_OPENMP)
+  list(APPEND LEGATE_CORE_CUDA_DEFS LEGATE_USE_OPENMP)
+endif()
+
+if(Legion_USE_GASNet)
+  list(APPEND LEGATE_CORE_CXX_DEFS LEGATE_USE_GASNET)
+  list(APPEND LEGATE_CORE_CUDA_DEFS LEGATE_USE_GASNET)
+endif()
+
+list(APPEND LEGATE_CORE_SOURCES
+  src/core/legate_c.cc
+  src/core/comm/comm.cc
+  src/core/data/allocator.cc
+  src/core/data/scalar.cc
+  src/core/data/store.cc
+  src/core/data/transform.cc
+  src/core/mapping/base_mapper.cc
+  src/core/mapping/core_mapper.cc
+  src/core/mapping/instance_manager.cc
+  src/core/mapping/mapping.cc
+  src/core/mapping/task.cc
+  src/core/runtime/context.cc
+  src/core/runtime/projection.cc
+  src/core/runtime/runtime.cc
+  src/core/runtime/shard.cc
+  src/core/task/return.cc
+  src/core/task/task.cc
+  src/core/utilities/debug.cc
+  src/core/utilities/deserializer.cc
+  src/core/utilities/machine.cc
+  src/core/utilities/linearize.cc
+)
+
+if(Legion_USE_CUDA)
+  list(APPEND LEGATE_CORE_SOURCES
+  src/core/comm/comm_nccl.cu
+  src/core/cuda/stream_pool.cu)
+endif()
+
+add_library(legate_core ${LEGATE_CORE_SOURCES})
+add_library(legate::core ALIAS legate_core)
+
+set_target_properties(legate_core
+           PROPERTIES EXPORT_NAME                         core
+                      LIBRARY_OUTPUT_NAME                 lgcore
+                      BUILD_RPATH                         "\$ORIGIN"
+                      INSTALL_RPATH                       "\$ORIGIN"
+                      CXX_STANDARD                        17
```

Comment:
I believe we don't want to go above CXX 14

Reply:
I seem to recall updating to CXX17 was discussed in a recent standup, though I can't find a record of it. I had set it to 14 initially, but updated to 17 after that call?

Reply:
We might want to check this again

Reply:
https://github.com/nv-legate/legate.core/blob/8d96152fff3152347a10572faa7e9fa6f42e90ef/install.py#L405-L407